File: TabuadaCompletaPersonalizada.py
while True:
    try:
        num = int(input("Qual número você quer ver a tabuada? "))
        print(f"\nTabuada do {num}:")
        for i in range(1, 11):
            print(f"{num} × {i:2} = {num*i:3}")
        print()
    except:
        print("Digite um número inteiro válido.")
    
    while True:
        continuar = input("Deseja ver a tabuada de outro número? (1 = sim / 0 = não): ")
        if continuar in ['0', '1']:
            break
        print("Digite apenas 0 ou 1.")
    
    if continuar == '0':
        print("Programa encerrado.")
        break

File: tabuadacompletapersonalizada.c
// exercicio3_tabuada_personalizada.c
#include <stdio.h>

int main() {
    int numero;
    int continuar;
    
    do {
        printf("\nDigite um numero para ver a tabuada: ");
        scanf("%d", &numero);
        
        printf("\nTabuada do %d:\n", numero);
        for(int i = 1; i <= 10; i++) {
            printf("%2d x %2d = %3d\n", i, numero, i * numero);
        }
        
        printf("\nDeseja ver a tabuada de outro numero? (1 = Sim / 0 = Nao): ");
        scanf("%d", &continuar);
        
    } while (continuar == 1);
    
    printf("Programa finalizado.\n");
    return 0;
}